mypyc/analysis.py

```diff
@@ -45,6 +45,9 @@ def get_cfg(blocks: List[BasicBlock]) -> CFG:
     pred_map = {}  # type: Dict[Label, List[Label]]
     exits = set()
     for block in blocks:
+        assert not any(isinstance(op, (Branch, Goto, Return)) for op in block.ops[:-1]), (
+            "Control-flow ops must be at the end of blocks")
```

Comment:
**Nit**: Maybe say something like `'Control-flow ops can exist only if they are the last instruction of a block'`? I'm not sure how to word it properly, but the `assert` is triggered when there exists a control-flow op as a non-last instruction, **not** when there is a lack of a control-flow op at the end of the block. 

The current error message makes it seem like the latter, which could be confusing to the user. (When I first saw this error message, I had to read the code before realizing that it was actually the former.)